daal4py/sklearn/test/test_run_to_run_stability_tests.py

```diff
@@ -14,322 +14,223 @@
-from sklearn.linear_model import LinearRegression
-from sklearn.linear_model import Ridge
-from sklearn.linear_model import ElasticNet
-from sklearn.linear_model import Lasso
-from sklearn.cluster import KMeans
-from sklearn.cluster import DBSCAN
+from sklearn.linear_model import LinearRegression, Ridge, ElasticNet, Lasso
+from sklearn.cluster import KMeans, DBSCAN
 from sklearn.decomposition import PCA
-
 from sklearn.svm import SVC
-from sklearn.datasets import make_classification
-from sklearn.datasets import make_regression
+from sklearn.datasets import make_classification, make_regression, make_blobs
+from sklearn.base import is_classifier, is_regressor
 
-##################################################
 
-def run_clf_test(func):
-    for features in [5, 10]:
-        X, y = make_classification(n_samples=4000, n_features=features, n_informative=features, n_redundant=0,
-            n_clusters_per_class=8, random_state=0)
-
-        baseline, name = func(X, y)
-
-        for i in range(10):
-            res, _ = func(X, y)
+# to reproduce errors even in CI
+d4p.daalinit(nthreads=100)
 
-            for a, b, n in zip(res, baseline, name):
-                np.testing.assert_allclose(a, b, rtol=0.0, atol=0.0, err_msg=str(n + " is incorrect"))
 
-def run_reg_test(func):
+def get_class_name(x):
+    return x.__class__.__name__
+
+
+WITHOUT_FIT = [
+    'DBSCAN',
+]
+
+
+def func(X, Y, model, methods):
+    clf = model
+    if get_class_name(model) not in WITHOUT_FIT:
+        clf.fit(X, Y)
+
+    res = []
+    name = []
+
+    for i in methods:
+        if i == 'predict':
+            res.append(clf.predict(X))
+            name.append(get_class_name(model) + '.predict(X)')
+        elif i == 'predict_proba':
+            res.append(clf.predict_proba(X))
+            name.append(get_class_name(model) + '.predict_proba(X)')
+        elif i == 'kneighbors':
+            dist, idx = clf.kneighbors(X)
+            res.append(dist)
+            name.append('dist')
+            res.append(idx)
+            name.append('idx')
+        elif i == 'fit_predict':
+            predict = clf.fit_predict(X)
+            res.append(predict)
+            name.append(get_class_name(model) + '.fit_predict')
+        elif i == 'transform':
+            res.append(clf.transform(X))
+            name.append(get_class_name(model) + '.transform(X)')
+        elif i == 'get_covariance':
+            res.append(clf.get_covariance())
+            name.append(get_class_name(model) + '.get_covariance()')
+        elif i == 'get_precision':
+            res.append(clf.get_precision())
+            name.append(get_class_name(model) + '.get_precision()')
+        elif i == 'score_samples':
+            res.append(clf.score_samples(X))
+            name.append(get_class_name(model) + '.score_samples(X)')
+
+    for i in clf.__dict__.keys():
+        ans = getattr(clf, i)
+        if isinstance(ans, (bool, float, int, np.ndarray, np.float64)):

```

Comment:
Are you sure that you cover all prev attributes by the list?

Reply:
yes, I've checked